packages/manager/modules/billing-components/src/components/services-actions/services-actions.controller.js

```diff
@@ -18,23 +18,23 @@ export default class ServicesActionsCtrl {
   }
 
   $onInit() {
-    let fetchAutoRenewLink = this.$q.when();
+    const fetchAutoRenewLink = this.$q.defer();
     if (!this.billingManagementAvailability) {
-      this.autorenewLink = null;
+      fetchAutoRenewLink.resolve(null);
     } else if (this.$injector.has('shellClient')) {
-      fetchAutoRenewLink = this.$injector
+      this.$injector
         .get('shellClient')
         .navigation.getURL('dedicated', '#/billing/autorenew')
         .then((url) => {
-          this.autorenewLink = url;
+          fetchAutoRenewLink.resolve(url);
         });
```

Comment:
If you defer a promise, it's better if you handle the error callback as well
```suggestion
        }).catch((error)=> fetchAutoRenewLink.reject(error));
```